addons/web/static/src/core/macro.js

```diff
@@ -174,24 +174,26 @@ export class MacroEngine {
         };
         this.observer = new MutationObserver(this.delayedCheck.bind(this));
         this.iframeObserver = new MutationObserver(() => {
-            const iframeEl = document.querySelector("iframe.o_iframe");
-            if (iframeEl) {
-                iframeEl.addEventListener("load", () => {
-                    if (iframeEl.contentDocument) {
-                        this.observer.observe(iframeEl.contentDocument, this.observerOptions);
-                    }
-                });
-                // If the iframe was added without a src, its load event was immediately fired and
-                // will not fire again unless another src is set. Unfortunately, the case of this
-                // happening and the iframe content being altered programmaticaly may happen.
-                // (E.g. at the moment this was written, the mass mailing editor iframe is added
-                // without src and its content rewritten immediately afterwards).
-                if (!iframeEl.src) {
+            [...document.querySelectorAll("iframe")].forEach((iframeEl) => {
+                const doIt = () => {
                     if (iframeEl.contentDocument) {
-                        this.observer.observe(iframeEl.contentDocument, this.observerOptions);
+                        iframeEl.contentDocument.addEventListener("readystatechange", (event) => {
+                            if (iframeEl.contentDocument.readyState === "complete") {
+                                this.delayedCheck();
+                                this.observer.observe(iframeEl.contentDocument, this.observerOptions);
+                            }
+                        });
+                        if (!iframeEl.src || iframeEl.contentDocument.readyState === "complete") {
```

Comment:
is `!iframe.src` still necessary?

Reply:
Unfortunatly yes, e.g. => test_template_edition